### PR TITLE
[wip] Rework spec for different JRuby implementation.

### PR DESCRIPTION
**Pull request by JonRowe**

### CODE REVIEW

**Comment by pirj** [comment]
@nevinera jfyi

File: spec/rspec/support/spec/stderr_splitter_spec.rb [hunk 1]
```diff
@@ -101,19 +101,19 @@
...
+      splitter.reopen(tempfile)
+      expect(splitter.to_io.path).to eq(tempfile.path)
     ensure
-      $stderr.reopen(cloned)
+      splitter.reopen(cloned)
       tempfile.close
       tempfile.unlink
     end
-    expect($stderr.to_io).not_to be_a(File)
+    expect(splitter.to_io.path).not_to eq tempfile.path
```
Comment (nevinera) [inline spec/rspec/support/spec/stderr_splitter_spec.rb:117]: While I think this is a valid test for the code in StderrSplitter, I don't think it exercises the case I was trying to rely on in https://github.com/rspec/rspec-expectations/pull/1460 anymore. Depending on whether we actually want to release that PR, that may be a good thing? I hadn't actually intended [this PR](https://github.com/rspec/rspec-support/pull/598) to be merged until we had consensus on that one.

File: spec/rspec/support/spec/stderr_splitter_spec.rb [hunk 2]
```diff
@@ -101,26 +101,28 @@
...
 
     cloned = splitter.clone
-    expect(splitter.to_io).not_to be_a(File)
+    expect(splitter.to_io).to be_stderr
 
     tempfile = Tempfile.new("foo")
     begin
       splitter.reopen(tempfile)
-      expect(splitter.to_io).to be_a(File)
+      expect(splitter.to_io).to_not be_stderr
```
Comment (JonRowe) [inline spec/rspec/support/spec/stderr_splitter_spec.rb:114]: @nevinera Unless I'm mistaken this represents the same effect as what you were testing, I'm not sure if I will merge this yet but it at least works for some JRuby

Comment (nevinera) [reply spec/rspec/support/spec/stderr_splitter_spec.rb:114]: Ah, I think the part I didn't follow was the bottom method - I assumed `be_stderr` was calling a `stderr?` method I wasn't aware of on the stream/splitter >.<

I'm not confident about how consistent that `inspect` is, but I'll trust your plan :-)

**Pull request closed without merging**